Do not assume a JobLogger receives a logger

wfarr opened the pull request:
In sidekiq/sidekiq#6200, the constructor for JobLogger was changed to accept an entire `Sidekiq::Config` rather than a logger.

To maintain compatibility with previous versions of Sidekiq, the constructor for Logcraft's Sidekiq::JobLogger has been changed to handle both cases by checking the type of the argument before assigning the logger.

wfarr committed:
Do not assume a JobLogger receives a logger

In sidekiq/sidekiq#6200, the constructor for JobLogger was changed to
accept an entire `Sidekiq::Config` rather than a logger.

To maintain compatibility with previous versions of Sidekiq, the
constructor for Logcraft's Sidekiq::JobLogger has been changed to handle
both cases by checking the type of the argument before assigning the
logger.

barna-emarsys commented:
@menyhertfatyol has tried this PR on one of our internal repositories, and has confirmed that it fixes compatibility with Sidekiq 7.3.0.

zormandi commented:
Thanks for the PR, @wfarr . I'm sorry I haven't had much time to deal with this and I see now that the tests are failing. It seems to be a trivial fix. I can take a look this weekend and hopefully merge this (and the other PR as well).

@barna-emarsys thank you for confirming. 👍

wfarr committed:
Add guard for older versions of sidekiq without Sidekiq::Config

It looks like older versions of sidekiq do not have this class defined,
so I've added a quick check that it is before we attempt to see if the
passed parameter to the job logger initializer is of that type.

wfarr commented:
> Thanks for the PR, @wfarr . I'm sorry I haven't had much time to deal with this and I see now that the tests are failing. It seems to be a trivial fix. I can take a look this weekend and hopefully merge this (and the other PR as well).

No worries at all! I definitely just tested this on Sidekiq 7.3, threw the PR out there, and assumed all would be good.

I've pushed a provisional change that I expect may fix the issue, but otherwise I'm happy to wait for you to take a look when time allows. 🙇

zormandi commented:
I ended up reimplementing this because I wanted to add some tests, add Sidekiq 7.3 to the CI matrix, etc. In any case, thanks a lot for contributing! ❤️
